### PR TITLE
[JUJU-1402] Search through existing relations when removing

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Previously, when checking for ambiguity in relations, juju would
check every potential relation, instead of looking through
relations that actually exist

Now query mongo to find all existing relations which match some
spec, and check for ambiguity by just checking the length of this slice

## Checklist

 - ~[ ] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - ~[ ] Added or updated [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) related to packages changed~
 - [ ] Comments answer the question of why design decisions were made

#### Deploy and relate some applications to create a potentially ambiguous relation
```
$ juju bootstrap lxd lxd
$ juju deploy grafana
$ juju deploy ceph-dashboard
$ juju relate grafana:dashboards ceph-dashboard:grafana-dashboard
$ juju status --relations
Model    Controller  Cloud/Region         Version   SLA          Timestamp
default  lxd         localhost/localhost  2.9.33.1  unsupported  17:53:18+01:00

App             Version  Status   Scale  Charm           Channel  Rev  Exposed  Message
ceph-dashboard           unknown      0  ceph-dashboard  stable     2  no       
grafana                  active       1  grafana         stable    56  no       Ready

Unit        Workload  Agent  Machine  Public address  Ports     Message
grafana/0*  active    idle   0        10.219.211.77   3000/tcp  Ready

Machine  State    Address        Inst id        Series  AZ  Message
0        started  10.219.211.77  juju-37d5c1-0  focal       Running

Relation provider                 Requirer            Interface          Type     Message
ceph-dashboard:grafana-dashboard  grafana:dashboards  grafana-dashboard  regular  
```
(NOTE: potentially re-establish relation between tests to restore status to the above

#### Attempt to remove w/o specifying relation
```
$ juju remove-relation grafana ceph-dashboard
$ echo $?
0
```
(then verify relation has been removed)

#### Attempt to remove with relations

```
$ juju remove-relation grafana:dashboards ceph-dashboard
```
and
```
$ juju remove-relation grafana ceph-dashboard:grafana-dashboard
```
and
```
$ juju remove-relation grafana:dashboards ceph-dashboard:grafana-dashboard
```

#### Attempt to remove a non-existing relation

```
$ juju remove-relation grafana ceph-dashboard:foo
ERROR relation "grafana ceph-dashboard:foo" not found (not found)
```
and
```
$ juju remove-relation grafana:bar ceph-dashboard
ERROR relation "grafana:bar ceph-dashboard" not found (not found)
```

#### Check attempt to remove ambiguous relation
```
$ juju relate grafana:website ceph-dashboard:prometheus
$ juju status
...
Relation provider                 Requirer                   Interface          Type     Message
ceph-dashboard:grafana-dashboard  grafana:dashboards         grafana-dashboard  regular  
grafana:website                   ceph-dashboard:prometheus  http               regular  

$ juju remove-relation grafana ceph-dashboard
ERROR ambiguous relation: "grafana ceph-dashboard" could refer to "grafana:dashboards ceph-dashboard:grafana-dashboard"; "ceph-dashboard:prometheus grafana:website"
```
then check it can be removed with:
```
$ juju remove-relation grafana:dashboards ceph-dashboard
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1964265

### CODE REVIEW

**Comment by hmlanigan** [comment]
@jack-w-shaw the qa steps frequently have 'relationship' rather than 'relation' in the commands and their output?

**Review by hmlanigan — APPROVED** [review]
a couple of nits, otherwise, lgtm

File: featuretests/cmd_juju_relation_test.go [hunk 1]
```diff
@@ -51,5 +51,5 @@ func (s *CmdRelationSuite) TestRemoveRelationSuccess(c *gc.C) {
 }
 
 func (s *CmdRelationSuite) TestRemoveRelationFail(c *gc.C) {
-	runCommandExpectFailure(c, "remove-relation", `relation "wordpress:db mysql:server" not found`, s.apps...)
+	runCommandExpectFailure(c, "remove-relation", `ERROR relation matching "wordpress mysql" not found (not found)`, s.apps...)
```
[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply featuretests/cmd_juju_relation_test.go:54]: Unfortunately there isn't much I can do about this, as this is just how the juju client prints "not found" errors. The previous test for a match, hence why a substring still succeeds.

[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply featuretests/cmd_juju_relation_test.go:54]: I think it's worth having the full error, as that is what other tests in this file do

**Pull request merged**